### PR TITLE
feat: add concurrency to configureAndRunAgents

### PR DESCRIPTION
**Pull request by fmartingr**
Add some concurrency to agent creation for large load tests

### CODE REVIEW

**Comment by agnivade** [comment]
This is a great improvement. Just take care to annotate every log line with the agent name, otherwise the log lines would get interleaved and might be difficult to debug in case of errors.

**Comment by agarciamontoro** [comment]
Oooh, looking good :eyes: Feel free to ask for a review when it's ready!

File: deployment/terraform/agent.go [hunk 1]
```diff
@@ -85,83 +86,113 @@ func (t *Terraform) configureAndRunAgents(extAgent *ssh.ExtAgent) error {
...
 
-		mlog.Info("Starting load-test api server")
-		if out, err := sshc.RunCommand("sudo systemctl daemon-reload && sudo service ltapi restart"); err != nil {
-			return fmt.Errorf("error running command, got output: %q: %w", out, err)
-		}
+	// Run another goroutine to check when all agents have finished configuring
+	go func() {
+		wg.Wait()
+		close(wgDone)
+	}()
```
Comment (agnivade) [inline deployment/terraform/agent.go:184]: I would restructure this a bit. Make `wg.Wait()` to be synchronous and remove the err chan. Closing the `errChan` while other goroutines are still sending data to it is actually a bug and will panic. 

What I'd suggest is:

```go
var wg sync.WaitGroup
for i, val := range agents {
  wg.Add(1)
  go func(i int, val string) {
    defer wg.Done()
    // do stuff here
    // log error as usual, but just set an atomic.Bool to true.
  }(i, val) // This is important to localize scope. Then you don't need the weird variable copying style.
}

wg.Wait()
if atomic.Bool == true, then return error. 
```

**Review by agnivade — APPROVED** [review]
Nicely done

File: deployment/terraform/agent.go [hunk 2]
```diff
@@ -85,83 +87,118 @@ func (t *Terraform) configureAndRunAgents(extAgent *ssh.ExtAgent) error {
...
+		agentNumber := i
+		instance := val
 
-		cmd := strings.Join(commands, " && ")
-		if out, err := sshc.RunCommand(cmd); err != nil {
-			return fmt.Errorf("error running command, got output: %q: %w", out, err)
-		}
+		go func() {
+			// Copy val into instance
+			defer wg.Done()
```
Comment (agarciamontoro) [inline deployment/terraform/agent.go:100]: Why is the `// Copy val into instance` comment inside the goroutine?

**Review by agarciamontoro — APPROVED** [review]
Looks amazing, thanks!

File: deployment/terraform/agent.go (hunk 2 above)
Comment (fmartingr) [reply deployment/terraform/agent.go:100]: Probably I moved some lines around during development. Let me remove it.

**Pull request merged**